enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,357 @@
+- Collect the relevant data from the ovn-controller logs and present it to the cluster administrator.  
+
+### Non-Goals
+
+- Network Policy Object Logging, `oc describe <entity>` takes care of that.
+
+## Proposal
+
+To begin implementing ACL audit logging, first the [network.operator.openshift.io](https://github.com/openshift/api/blob/master/operator/v1/types_network.go) API needs to be updated.  These changes involve adding the optional `aclLoggingRateLimit` flag to the `OVNKubernetesConfig` struct as follows.
+
+```go
+// ovnKubernetesConfig contains the configuration parameters for networks
+// using the ovn-kubernetes network project
+type OVNKubernetesConfig struct {
+ // mtu is the MTU to use for the tunnel interface. This must be 100
+ // bytes smaller than the uplink mtu.
+ // Default is 1400
+ // +kubebuilder:validation:Minimum=0
+ // +optional
+ MTU *uint32 `json:"mtu,omitempty"`
+ // geneve port is the UDP port to be used by geneve encapulation.
+ // Default is 6081
+ // +kubebuilder:validation:Minimum=1
+ // +optional
+ GenevePort *uint32 `json:"genevePort,omitempty"`
+ // HybridOverlayConfig configures an additional overlay network for peers that are
+ // not using OVN.
+ // +optional
+ HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+ // ipsecConfig enables and configures IPsec for pods on the pod network within the
+ // cluster.
+ // +optional
+ IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
+ 
+ <--- BEGIN NEW CODE --->
+ // aclLoggingRateLimit allows for the configuring of max acl logging rate
+ // defaults to 20 messages per second
+ // +optional
+ aclLoggingRateLimit  *uint32 `json:"aclLoggingRateLimit,omitempty"`
+ <--- END NEW CODE --->
+
+}
+```
+
+The flag `aclLoggingRateLimit` defaults to `20` messages per second.
+
+This will result in an updated OVNKubernetes Operator configuration object: `(.spec.defaultNetwork.ovnKubernetesConfig)` which will enable cluster-wide configuration of ACL logging at cluster installation time:
+
+```go
+spec:
+  defaultNetwork:
+    type: OVNKubernetes
+    ovnKubernetesConfig:
+      mtu: 1400
+      genevePort: 6081
+      ipsecConfig: {}
+      aclLoggingRateLimit: 20
+```
+
+To enable the ACL logging on a per namespace basis using metadata annotations in the namespace's definition like the following
+
+```go
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: tenantA
+  annotations:
+    k8s.ovn.org/acl-logging: '{ "deny": "alert", "allow": "notice" }'
+```
+
+or
+
+```go
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: tenantB
+  annotations:
+    k8s.ovn.org/acl-logging: '{ "deny": "notice" }'
+```
+
+The logging can be activated for either allow, drop, or allow and drop actions. The severity must be one of `alert`, `warning`,
+`notice`, `info`, or `debug` as described in [OVN documentation](http://www.openvswitch.org/support/dist-docs/ovn-nbctl.8.html).
+
+each level is listed in order or descending severity
+
+```go
```

Comment:
These values aren't super interesting, since that's just an ovn convention. The severity is just plumbed through from the vswitch to the output log. I would remove this section.

Reply:
Sounds good to me, I will just say "Refer to ovs documentation" or something along those lines

Reply:
No, that doesn't matter; we generate the severity. It would be something consumed by the log aggregator.